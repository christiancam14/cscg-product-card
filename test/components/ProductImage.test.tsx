import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product1, product2 } from '../data/products';

describe('Pruebas en ProductImage', () => {
  test('Debe aparecer el titulo', () => {
    const wrapper = renderer.create(<ProductImage img={product2.img} />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Debe mostrar el componente con el nombre del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => <ProductImage img={product2.img} />}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
