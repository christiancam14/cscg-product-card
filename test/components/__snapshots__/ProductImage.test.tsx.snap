// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductImage Debe aparecer el titulo 1`] = `
<img
  alt="Coffee mug"
  className="productImg undefined"
  src="./coffee-mug.png"
/>
`;

exports[`Pruebas en ProductImage Debe mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffee mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
